Amplitude/Amplitude.m

```diff
@@ -893,6 +893,7 @@ - (void)uploadEventsWithLimit:(int) limit
         long numEvents = limit > 0 ? fminl(eventCount, limit) : eventCount;
         if (numEvents == 0) {
             self->_updatingCurrently = NO;
+            [self endBackgroundTaskIfNeeded];
```

Comment:
here is example case, when numEvents == 0, and the background task will never ends